docs/guides/prefect-deploy.md

```diff
@@ -0,0 +1,628 @@
+    To expand environment variables in your shell script, set `expand_env_vars: true` in your `run_shell_script` step. For example:
+
+    ```yaml
+    - prefect.deployments.steps.run_shell_script:
+        id: get-user
+        script: echo $USER
+        stream_output: true
+        expand_env_vars: true
+    ```
+
+    Without `expand_env_vars: true`, the above step would return a literal string `$USER` instead of the current user.
+
+- `pip_install_requirements` installs dependencies from a `requirements.txt` file within a specified directory.
+
+Below is an example of installing dependencies from a `requirements.txt` file after cloning:
+
+```yaml
+pull:
+    - prefect.deployments.steps.git_clone:
+        id: clone-step
+        repository: https://github.com/org/repo.git
+    - prefect.deployments.steps.pip_install_requirements:
+        directory: {{ clone-step.directory }}
+        requirements_file: requirements.txt
+        stream_output: False
+```
+
+Below is an example that retrieves an access token from a 3rd party Key Vault and uses it in a private clone step:
+
+```yaml
+pull:
+- prefect.deployments.steps.run_shell_script:
+    id: get-access-token
+    script: az keyvault secret show --name <secret name> --vault-name <secret vault> --query "value" --output tsv
+    stream_output: false
+- prefect.deployments.steps.git_clone:
+    repository: https://bitbucket.org/samples/deployments.git
+    branch: master
+    access_token: "{{ get-access-token.stdout }}"
+```
+
+You can also run custom steps by packaging them. In the example below, `retrieve_secrets` is a custom python module that has been packaged into the default working directory of a docker image (which is /opt/prefect by default). `main` is the function entry point, which returns an access token (e.g. `return {"access_token": access_token}`) like the preceding example, but utilizing the Azure Python SDK for retrieval.
+
+```yaml
+- retrieve_secrets.main:
+    id: get-access-token
+- prefect.deployments.steps.git_clone:
+    repository: https://bitbucket.org/samples/deployments.git
+    branch: master
+    access_token: '{{ get-access-token.access_token }}'
+```
+
+### Templating Options
+
+Values that you place within your `prefect.yaml` file can reference dynamic values in several different ways:
+
+- **step outputs**: every step of both `build` and `push` produce named fields such as `image_name`; you can reference these fields within `prefect.yaml` and `prefect deploy` will populate them with each call.  References must be enclosed in double brackets and be of the form `"{{ field_name }}"`
+- **blocks**: [Prefect blocks](/concepts/blocks) can also be referenced with the special syntax `{{ prefect.blocks.block_type.block_slug }}`; it is highly recommended that you use block references for any sensitive information (such as a GitHub access token or any credentials) to avoid hardcoding these values in plaintext
```

Comment:
```suggestion
- **blocks**: [Prefect blocks](/concepts/blocks) can also be referenced with the special syntax `{{ prefect.blocks.block_type.block_slug }}`. It is highly recommended that you use block references for any sensitive information (such as a GitHub access token or any credentials) to avoid hardcoding these values in plaintext
```